pydantic_ai_slim/pydantic_ai/ui/ag_ui/_event_stream.py

```diff
@@ -86,10 +87,18 @@ def content_type(self) -> str:
     def encode_event(self, event: BaseEvent) -> str:
         return self._event_encoder.encode(event)
 
+    async def handle_event(self, event: NativeEvent) -> AsyncIterator[BaseEvent]:
+        """Override to set timestamps on all AG-UI events."""
+        async for agui_event in super().handle_event(event):
+            if agui_event.timestamp is None:
+                agui_event.timestamp = int(now_utc().timestamp() * 1000)
```

Comment:
Does AG-UI document this should be in ms precision?

Reply:
Yeah, I found it in a fork, updated the link to the official repo -- still a community provided SDK

Reply:
@thejens Ok that's good enough for me, thanks for double checking!